src/main/java/org/opentripplanner/framework/time/OffsetDateTimeParser.java

```diff
@@ -0,0 +1,42 @@
+package org.opentripplanner.framework.time;
+
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.format.DateTimeParseException;
+import java.util.Optional;
+
+public class OffsetDateTimeParser {
+
+  /**
+   * We need to have two offsets, in order to parse both "+0200" and "+02:00". The first is not
+   * really ISO-8601 compatible with the extended date and time. We need to make parsing strict, in
+   * order to keep the minute mandatory, otherwise we would be left with an unparsed minute
+   */
+  public static final DateTimeFormatter LENIENT_PARSER = new DateTimeFormatterBuilder()
+    .parseCaseInsensitive()
+    .parseLenient()
+    .append(DateTimeFormatter.ISO_LOCAL_DATE_TIME)
+    .optionalStart()
+    .parseStrict()
+    .appendOffset("+HH:MM:ss", "Z")
+    .parseLenient()
+    .optionalEnd()
+    .optionalStart()
+    .appendOffset("+HHmmss", "Z")
+    .optionalEnd()
+    .toFormatter();
```

Comment:
This feels like a weirdly long set of method calls where something gets parsed multiple times. I don't know if there would be a more readable solution to this.

Reply:
In the dev meeting we decided that we are okay with this.